### PR TITLE
[WebKit] Add support for Xcode 15

### PR DESCRIPTION
**Pull request by haritha-mohan**

### CODE REVIEW

File: tests/xtro-sharpie/macOS-WebKit.ignore [hunk 1]
```diff
@@ -2,3 +2,46 @@
...
 !unknown-native-enum! WKSelectionGranularity bound
+
+# xcode 15
+# API has been deprecated in macOS..
+!incorrect-protocol-member! WebOpenPanelResultListener::cancel is REQUIRED and should be abstract
+!incorrect-protocol-member! WebOpenPanelResultListener::chooseFilename: is REQUIRED and should be abstract
+!incorrect-protocol-member! WebOpenPanelResultListener::chooseFilenames: is REQUIRED and should be abstract
+!incorrect-protocol-member! WebPolicyDecisionListener::download is REQUIRED and should be abstract
+!incorrect-protocol-member! WebPolicyDecisionListener::ignore is REQUIRED and should be abstract
+!incorrect-protocol-member! WebPolicyDecisionListener::use is REQUIRED and should be abstract
```
Comment (haritha-mohan) [inline tests/xtro-sharpie/macOS-WebKit.ignore:13]: https://developer.apple.com/documentation/webkit/webopenpanelresultlistener?language=objc
The API has been deprecated, so assuming we don’t need to modify the Abstract attrs?

File: tests/xtro-sharpie/macOS-WebKit.ignore [hunk 2]
```diff
@@ -2,3 +2,46 @@
...
+
+# older/legacy API
+!missing-selector! +NSObject::isKeyExcludedFromWebScript: not bound
```
Comment (haritha-mohan) [inline tests/xtro-sharpie/macOS-WebKit.ignore:16]: Support for these API were added as part of macOS 10+ and is noted as legacy on Apple Docs..do we want to add support for these right now?
https://developer.apple.com/documentation/objectivec/nsobject/webplugincontainer?language=objc

File: tests/xtro-sharpie/api-annotations-dotnet/macOS-WebKit.ignore [hunk 3]
```diff
@@ -1,3 +1,39 @@
...
+!missing-selector! NSObject::webPlugInInitialize not bound
+!missing-selector! NSObject::webPlugInMainResourceDidFailWithError: not bound
+!missing-selector! NSObject::webPlugInMainResourceDidFinishLoading not bound
+!missing-selector! NSObject::webPlugInMainResourceDidReceiveData: not bound
+!missing-selector! NSObject::webPlugInMainResourceDidReceiveResponse: not bound
+!missing-selector! NSObject::webPlugInSetIsSelected: not bound
+!missing-selector! NSObject::webPlugInStart not bound
+!missing-selector! NSObject::webPlugInStop not bound
+
+# is bound, not sure what xtro is on
```
Comment (tj-devel709) [inline tests/xtro-sharpie/api-annotations-dotnet/macOS-WebKit.ignore:30]: 😂

**Review by tj-devel709 — APPROVED** [review]

File: src/webkit.cs [hunk 4]
```diff
@@ -5549,6 +5639,14 @@ interface WKWebViewConfiguration : NSCopying, NSSecureCoding {
 		[Mac (12, 0), iOS (15, 0), MacCatalyst (15, 0), NoTV]
 		[Export ("upgradeKnownHostsToHTTPS")]
 		bool UpgradeKnownHostsToHttps { get; set; }
+
+		[Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+		[Export ("allowsInlinePredictions")]
+		bool AllowsInlinePredictions { get; set; }
+
+		[NoiOS, Mac (14, 0)]
```
Comment (haritha-mohan) [inline src/webkit.cs:5647]:
```suggestion
		[NoiOS, Mac (14, 0), NoMacCatalyst]
```

File: tests/xtro-sharpie/macOS-WebKit.ignore (hunk 2 above)
Comment (rolfbjarne) [reply tests/xtro-sharpie/macOS-WebKit.ignore:16]: Sounds good to not do anything for now.

**Review by rolfbjarne — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

File: tests/xtro-sharpie/api-annotations-dotnet/common-WebKit.ignore [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# xcode 15
+# requires OS_nw_proxy_config type which has not been bound yet, part of Xcode 15 Network binding
+!missing-selector! WKWebsiteDataStore::proxyConfigurations not bound
+!missing-selector! WKWebsiteDataStore::setProxyConfigurations: not bound
```
Comment (mandel-macaque) [inline tests/xtro-sharpie/api-annotations-dotnet/common-WebKit.ignore:4]: @haritha-mohan  can you please create an issue to track this, I'll take care of it once I do the network framework.

**Review by mandel-macaque — APPROVED** [review]

File: tests/xtro-sharpie/api-annotations-dotnet/common-WebKit.ignore (hunk 5 above)
Comment (haritha-mohan) [reply tests/xtro-sharpie/api-annotations-dotnet/common-WebKit.ignore:4]: 👍 https://github.com/xamarin/xamarin-macios/issues/19259

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**